### PR TITLE
Add `--pruning` option to `pytorch_lightning_simple.py`

### PR DESCRIPTION
**Pull request by yutayamazaki**
- Related issue #731

### CODE REVIEW

File: examples/pytorch_lightning_simple.py [hunk 1]
```diff
@@ -165,7 +165,15 @@ def objective(trial):
 
 
 if __name__ == '__main__':
-    study = optuna.create_study(direction='maximize')
+    parser = argparse.ArgumentParser(description='PyTorch Lightning example.')
+    parser.add_argument('--pruning', '-p', action='store_true',
+                        help='Activate the pruning feature. `MedianPruner` stops unpromising '
+                             'trials at the early stages of training.')
+    args = parser.parse_args()
+    args.pruning = True
```
Comment (crcrpar) [inline examples/pytorch_lightning_simple.py]: I think this line should be removed because this seems to always enable pruning.
```suggestion
```

Comment (yutayamazaki) [reply examples/pytorch_lightning_simple.py]: I forgot to delete the line used for debugging, so I fixed it.

**Comment by sile** [comment]
Thank you for your PR.

I have some comments:
- To enable the pruning feature, it's needed to report intermediate values by using [PyTorchLightningPruningCallback](https://optuna.readthedocs.io/en/stable/reference/integration.html#optuna.integration.PyTorchLightningPruningCallback).
- Please add this example to the [Examples of Pruning](https://github.com/optuna/optuna/blob/master/examples/README.md#examples-of-pruning) list in `examples/README.md`.

**Comment by hvy** [comment]
Thanks for the PR! I was trying this out locally but it seems like the accuracy value isn't passed to the pruner through the log. 🤔

**Comment by toshihikoyanase** [comment]
> Thanks for the PR! I was trying this out locally but it seems like the accuracy value isn't passed to the pruner through the log meaning the pruning doesn't work. 🤔 Could you take a look?

I found that the accuracy values were correctly passed to the pruner when I removed `train_percent_check` and `val_percent_check` from the options of `pl.Trainer`. Maybe this is an issue of pytorch-lightning,  so how about stop using those options?

**Comment by hvy** [comment]
@yutayamazaki Do you have any updates on this PR? Otherwise if you are busy, we could take over it and continue :)

**Comment by yutayamazaki** [comment]
> Thanks for the quick fix, LGTM!

Sorry it took so long to address to the problem!!

**Comment by hvy** [comment]
Thanks a lot for the long running PR, and sorry for the delay. We had a short code freeze period before the release of v1.0. LGTM!

**Pull request merged**